src/iptvsimple/utilities/StreamUtils.cpp

```diff
@@ -63,7 +63,41 @@ void StreamUtils::SetAllStreamProperties(PVR_NAMED_VALUE* properties, unsigned i
     }
     else // inputstream.adpative
     {
-      StreamUtils::SetStreamProperty(properties, propertiesCount, propertiesMax, PVR_STREAM_PROPERTY_STREAMURL, streamURL);
+      // If no media headers are explicitly set for inputstream.adaptive,
+      // strip the headers from streamURL and put it to media headers property
+
+      if (channel.GetProperty("inputstream.adaptive.stream_headers").empty()) {
+        // preserve any kodi protocol options after "|"
+        std::string url = streamURL;
+        std::string protocolOptions;
+        std::string untrimmed_header = "";
+
+        size_t found = streamURL.find_first_of('|');
+        if (found != std::string::npos) {
+          url = streamURL.substr(0, found);
+          protocolOptions = streamURL.substr(found + 1, streamURL.length());
+
+          std::vector<std::string> headers = StringUtils::Split(protocolOptions, "&");
+          protocolOptions.clear();
+          for (std::vector<std::string>::iterator b(headers.begin()), e(headers.end()); b != e; ++b) {
+            std::string::size_type pos(b->find('='));
+            if(pos != std::string::npos)
+              untrimmed_header = b->substr(0, pos);
+              protocolOptions += StringUtils::Trim(untrimmed_header);
+              untrimmed_header = b->substr(pos+1);
+              protocolOptions += "=" + WebUtils::UrlEncode(StringUtils::Trim(untrimmed_header)) + "&";
+          }
+
+          protocolOptions.pop_back();
+
+          StreamUtils::SetStreamProperty(properties, propertiesCount, propertiesMax, PVR_STREAM_PROPERTY_STREAMURL, url);
+          StreamUtils::SetStreamProperty(properties, propertiesCount, propertiesMax, "inputstream.adaptive.stream_headers", protocolOptions);
+        } else {
+          StreamUtils::SetStreamProperty(properties, propertiesCount, propertiesMax, PVR_STREAM_PROPERTY_STREAMURL, streamURL);
+        }
+      } else
```

Comment:
If the rest of the if/else has brackets don't leave one hanging like this ;)

Also, I don't think you need this line repeated. Just use a bool to see if the stream_headers were set and then only do this one if they were not.

Reply:
done